src/plan_expand_hypertable.c

```diff
@@ -945,11 +954,56 @@ get_explicit_chunk_oids(CollectQualCtx *ctx, Hypertable *ht)
 								NameStr(ht->fd.table_name))));
 
 			chunk_oids = lappend_int(chunk_oids, chunk->table_id);
+
+			num_chunks++;
+
+			if (chunks == NULL)
+			{
+				Assert(num_chunks == 1);
+				chunks = (Chunk **) palloc0(sizeof(Chunk *));
+			}
+			else
+				chunks = (Chunk **) repalloc(chunks, sizeof(Chunk *) * num_chunks);
```

Comment:
It looks like this will reallocate memory for every chunk. Since the number of chunks is known a priori, I suggest allocating the exact size array outside the loop:

```
ArrayType *arr = DatumGetArrayTypeP(chunks_arg->constvalue);

if (ARR_NDIM(arr) != 1)
    ereport(ERROR,
                 (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
                  errmsg("invalid number of array dimensions for chunks_in")));

chunks = palloc0(sizeof(Chunk *) * ArrayGetNItems(ARR_NDIM(arr), ARR_DIMS(arr));
```

Reply:
The while loop checks for `isnull` for each entry in the array. That's the reason I did a repalloc. But yeah, can palloc like above and adjust the `num_chunks` in the loop appropriately

Reply:
Also, `isnull=true` is an error condition so `num_chunks` shouldn't need adjustment. It might still make sense to increment it in the loop though.